### DISCUSSION TITLE
Suggestion for next awesome upgrade

### DISCUSSION THREAD

**Discussion by JamesPlayfair** [opensteno/plover#1394] [Ideas]
The one feature that would take plover to a whole new level and really sell open steno to the world would be a recursive 'templating' feature.   i.e.  The stroke translations in the dictionaries could include placeholder fields where additional strokes would be accepted at execution time.    This would provide extreme efficiencies when entering structured/hierarchichal data such as program code, html etc.

**Comment by user202729** [comment]
I don't really understand the idea but you can already use Plover with classical snippet-expansion programs. (such as UltiSnips in vim, etc.)

Also, I have plugin which perhaps the last word into a pair of parentheses, so I can type [function] [value]  [wrap] and it will result in `function(value)`. I could as well extend it to enter expression with full RPN syntax, but it doesn't feel very necessary.
